flink-connectors/flink-connector-kafka/src/test/java/org/apache/flink/streaming/connectors/kafka/KafkaTestBase.java

```diff
@@ -79,7 +79,8 @@
  */
 @SuppressWarnings("serial")
 @RetryOnFailure(times = 3)
```

Comment:
In javadoc[1] for `RetryOnFailure` is stated that 
> \* <p>Add the {@link RetryRule} to your test class and annotate the class and/or tests with {@link
> \* RetryOnFailure}.

which means it makes sense to use it only with junit4. With junit5 there should be used `RetryOnException`

[1] https://github.com/apache/flink/blob/f8c6a668cd2b887f33a0cf4608de2d6b95c71f03/flink-test-utils-parent/flink-test-utils-junit/src/main/java/org/apache/flink/testutils/junit/RetryOnFailure.java#L30

Reply:
Hi @snuyanzin 

After going through the `RetryOnException` java docs it also mentions the same thing

> Annotation to use with {@link org.apache.flink.testutils.junit.RetryRule}.
> 
> <p>Add the {@link org.apache.flink.testutils.junit.RetryRule} to your test class and annotate the
>  class and/or tests with {@link RetryOnException}.

I also went through the `RetryExtension` source code and looks like it reads from both the annotations (RetryOnException, RetryOnFailure)

Do you still think it makes sense to migrate to `RetryOnException` annotation?

Reply:
yes it makes sense
it will simplify to remove junit4 related stuff